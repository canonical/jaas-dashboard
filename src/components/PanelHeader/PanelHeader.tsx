import type { ReactNode } from "react";

type Props = {
  id: string;
  title: ReactNode;
  // TODO: Make onRemovePanelQueryParams required and modify tests.
  onRemovePanelQueryParams?: () => void;
};

export default function PanelHeader({
  title,
  id,
  onRemovePanelQueryParams,
}: Props): JSX.Element {
  return (
    <div className="p-panel__header">
      <div className="p-panel__title" id={id}>
        {title}
      </div>
      <div className="p-panel__controls">
        <button
          onClick={onRemovePanelQueryParams}
          className="p-button--base js-aside-close u-no-margin--bottom has-icon"
        >
          <i className="p-icon--close"></i>
        </button>
      </div>
    </div>
  );
}
