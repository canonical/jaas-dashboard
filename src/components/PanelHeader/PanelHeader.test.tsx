import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import PanelHeader from "./PanelHeader";

describe("PanelHeader", () => {
  it("Renders the supplied title", () => {
    const title = "My Title";
    render(
      <MemoryRouter
        initialEntries={["/models/user-eggman@external/new-search-aggregate"]}
      >
        <PanelHeader id="123" title={title} />
      </MemoryRouter>
    );
    expect(screen.getByText(title)).toHaveClass("p-panel__title");
  });

  it("Removes all query params when close button clicked", async () => {
    window.history.pushState({}, "", "/models?model=cmr&panel=share-model");
    render(
      <BrowserRouter>
        <PanelHeader id="123" title="Title" />
      </BrowserRouter>
    );
    const searchParams = new URLSearchParams(window.location.search);
    expect(searchParams.get("panel")).toEqual("share-model");
    expect(searchParams.get("model")).toEqual("cmr");
    await userEvent.click(screen.getByRole("button"));
    const searchParamsAfterClose = new URLSearchParams(window.location.search);
    expect(searchParamsAfterClose.get("panel")).toEqual(null);
    expect(searchParamsAfterClose.get("model")).toEqual(null);
  });
});
