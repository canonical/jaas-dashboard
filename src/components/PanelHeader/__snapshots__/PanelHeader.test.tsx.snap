// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader should render successfully 1`] = `
<div>
  <div
    class="p-panel__header"
  >
    <div
      class="p-panel__title"
      id="mockID"
    >
      Title
    </div>
    <div
      class="p-panel__controls"
    >
      <button
        class="p-button--base js-aside-close u-no-margin--bottom has-icon"
      >
        <i
          class="p-icon--close"
        />
      </button>
    </div>
  </div>
</div>
`;
